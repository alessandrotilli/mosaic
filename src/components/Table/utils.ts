export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
