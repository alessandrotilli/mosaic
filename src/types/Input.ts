import { IBase, ILocalizable } from "./Base";

export enum InputSize {
  default = "medium",
  small = "small",
}

export enum InputDataType {
  default = "text",
  number = "number",
  password = "password",
}

export enum InputVariant {
  default = "outlined",
  filled = "filled",
  underlined = "standard",
}

export interface IInput extends ILocalizable {
  disabled?: boolean;
  label?: string;
  onChange?: Function;
  placeholder?: string;
  required?: boolean;
  shrink?: boolean;
  size?: InputSize;
  type?: InputDataType;
  variant?: InputVariant;
}
