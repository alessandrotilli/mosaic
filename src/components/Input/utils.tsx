import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
