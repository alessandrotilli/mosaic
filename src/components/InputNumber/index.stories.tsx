import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, select, number, text } from "@storybook/addon-knobs";
import { InputSize, InputVariant } from "../../types/Input";
import FormMock from "../../utils/mocks/FormMock";
import IntlProviderMock, { LocaleMock, MessageMock } from "../../utils/mocks/IntlProviderMock";
import { getDocumentationPage, StoriesWrapper } from "../../utils/stories";
import InputNumber, { InputNumberWithProps, DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS } from ".";

export default {
  title: "InputNumber",
  component: InputNumberWithProps,
  parameters: {
    ...getDocumentationPage({
      basedOn: "@material-ui/core/TextField",
      component: "InputNumber",
      e2eTestInfo: {
        dataCyDefault: DATA_CY_DEFAULT,
        dataCyShortcut: DATA_CY_SHORTCUT,
      },
      localizableProps: LOCALIZABLE_PROPS,
    }),
  },
};

export const Canvas = () => (
  // FormMock simulates external form component handling state
  // In a real case scenario "onChange" and "value" props must be passed to InputNumber
  <FormMock inputValue={number("value", 5)} onInputChange={action("onChange callback")}>
    <InputNumber
      dataCy={text("dataCy", DATA_CY_DEFAULT)}
      disabled={boolean("disabled", false)}
      integer={boolean("integer", true)}
      label={text("label", "Label")}
      localized={boolean("localized", false)}
      minValue={number("minValue", 0)}
      maxValue={number("maxValue", 9)}
      onChange={action("onChange callback")}
      placeholder={text("placeholder", "Placeholder Text")}
      required={boolean("required", false)}
      shrink={boolean("shrink", true)}
      size={select("size", InputSize, InputSize.default)}
      variant={select("variant", InputVariant, InputVariant.default)}
      value={number("value", 5)}
    />
  </FormMock>
);

export const Bounded = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Bounded 0-9" minValue={0} maxValue={9} required value={5} />
  </StoriesWrapper>
);

export const DefaultNull = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Default null" />
  </StoriesWrapper>
);

export const IntegerAndFloat = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Integer" value={5} />
    <InputNumber dataCy="input-number" integer={false} label="Float" value={5.5} />
  </StoriesWrapper>
);

export const Localized = () => (
  // IntlProviderMock simulates external IntlProvider context
  <StoriesWrapper>
    <IntlProviderMock locale={select("locale", LocaleMock, LocaleMock.en)}>
      <InputNumber
        label={MessageMock.inputNumber}
        localized
        onChange={action("Click on Button")}
        placeholder={MessageMock.inputNumber}
      />
    </IntlProviderMock>
  </StoriesWrapper>
);

export const Required = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Required" required value={10} />
  </StoriesWrapper>
);

export const Size = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Default" value={10} />
    <InputNumber dataCy="input-number" label="Small" size={InputSize.small} value={10} />
  </StoriesWrapper>
);

export const Variant = () => (
  <StoriesWrapper>
    <InputNumber dataCy="input-number" label="Default" value={10} />
    <InputNumber dataCy="input-number" label="Filled" variant={InputVariant.filled} value={10} />
  </StoriesWrapper>
);
