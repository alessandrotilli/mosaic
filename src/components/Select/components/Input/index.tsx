import React, { CSSProperties, FC, useMemo } from "react";
import {
  AutocompleteRenderInputParams as MUIAutocompleteRenderInputParams,
  Skeleton as MUISkeleton,
  TextField as MUITextField,
} from "@mui/material";

import { ILoadable } from "../../../../types/Base";
import { IInputField } from "../../../../types/Input";
import { getComposedDataCy, ISubpart } from "../../../../utils";

type ISelectInputField = Pick<IInputField, "dataCy" | "label" | "placeholder"> &
  Required<Pick<IInputField, "required" | "size" | "type" | "variant">>;

interface ISelectInput extends ISelectInputField, ILoadable {
  forwarded: MUIAutocompleteRenderInputParams;
}

export const SELECT_LOADING_SUBPART: ISubpart = {
  label: "Loading",
};

const SelectInput: FC<ISelectInput> = ({
  dataCy = "select-input",
  forwarded,
  label,
  loading,
  placeholder,
  required,
  size,
  style: externalStyle,
  type,
  variant,
}) => {
  const { inputProps: forwardedInputProps, ...forwardedRefs } = forwarded;

  const style = useMemo((): CSSProperties => ({ ...externalStyle, width: "100%" }), [externalStyle]);

  const inputDataCy = useMemo(
    () => (!loading ? dataCy : getComposedDataCy(dataCy, SELECT_LOADING_SUBPART)),
    [dataCy, loading]
  );

  const inputProps = useMemo(
    () => ({
      ...forwardedInputProps,
      "data-cy": inputDataCy,
      style,
    }),
    [forwardedInputProps, inputDataCy, style]
  );

  if (loading) {
    return (
      <MUISkeleton width="100%">
        <MUITextField
          {...forwardedRefs}
          inputProps={inputProps}
          margin="normal"
          size={size}
          variant={variant}
          style={style}
        />
      </MUISkeleton>
    );
  }

  return (
    <MUITextField
      {...forwardedRefs}
      inputProps={inputProps}
      label={label}
      margin="normal"
      placeholder={placeholder}
      required={required}
      size={size}
      style={style}
      type={type}
      variant={variant}
    />
  );
};

export default SelectInput;
