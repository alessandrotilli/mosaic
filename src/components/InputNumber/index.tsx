import React, { FC } from "react";
import { InputDataType, InputSize, InputVariant } from "../../types/Input";
import { IInputNumber } from "../../types/InputNumber";
import { StyledMUITextField } from "./styled";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";

const getControlledValue = (value: number | null): any => {
  return value === null ? "" : value;
};

const getNumericValue = (value: string, options: any): number | null => {
  const { integer, minValue, maxValue } = options;
  const numericValue = integer ? parseInt(value, 10) : parseFloat(value);
  if (isNaN(numericValue)) {
    return null;
  }

  if (numericValue >= minValue && numericValue <= maxValue) {
    return numericValue;
  }

  return numericValue < minValue ? minValue : maxValue;
};

export const DATA_CY_DEFAULT = "input-number";
export const DATA_CY_SHORTCUT = "label";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [
  { name: "label", type: "string" },
  { name: "placeholder", type: "string" },
];

const InputNumber: FC<IInputNumber> = ({
  dataCy = DATA_CY_DEFAULT,
  disabled = false,
  integer = true,
  label,
  minValue = Number.MIN_SAFE_INTEGER,
  maxValue = Number.MAX_SAFE_INTEGER,
  onChange = undefined,
  placeholder = undefined,
  required = false,
  shrink = true,
  size = InputSize.default,
  value = null,
  variant = InputVariant.default,
}) => {
  const onChangeHandler = (event: any) => {
    const numericValue = getNumericValue(event.target.value, { integer, minValue, maxValue });
    if (onChange) {
      onChange(numericValue);
    }
  };

  return (
    <StyledMUITextField
      disabled={disabled}
      InputLabelProps={{
        shrink,
      }}
      inputProps={{
        "data-cy": dataCy,
        style: { textAlign: "right" },
      }}
      InputProps={{
        readOnly: disabled,
      }}
      label={label}
      margin="normal"
      onChange={onChangeHandler}
      placeholder={placeholder}
      required={required}
      size={size}
      type={InputDataType.number}
      variant={variant}
      value={getControlledValue(value)}
    />
  );
};

export const InputNumberWithProps = InputNumber;

export default localized(InputNumber, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});
