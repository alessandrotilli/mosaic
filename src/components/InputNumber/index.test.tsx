import renderer from "react-test-renderer";

import { InputSize, InputVariant } from "../../types/Input";
import { IInputNumber } from "../../types/InputNumber";
import { MessageMock } from "../../utils/mocks/IntlProviderMock";
import { getLocalizedTestable } from "../../utils/tests";

import InputNumber, { DATA_CY_DEFAULT } from ".";

const defaultProps: IInputNumber = {};

const getInputNumberTestable = (props?: IInputNumber, dataCy = DATA_CY_DEFAULT) =>
  getLocalizedTestable(InputNumber, { dataCy, domNode: "input", props: { ...defaultProps, ...props } });

describe("InputNumber test suite:", () => {
  it("default", () => {
    const { element, wrapper } = getInputNumberTestable();
    expect(wrapper).toHaveLength(1);
    expect(wrapper.hasClass("MuiOutlinedInput-input"));
    expect(wrapper.prop("disabled")).toBeFalsy();
    expect(wrapper.prop("required")).toBeFalsy();
    expect(wrapper.prop("type")).toEqual("number");

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("dataCy", () => {
    const { element, wrapper } = getInputNumberTestable({ dataCy: "custom" }, "custom");
    expect(wrapper).toHaveLength(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  // TODO: improve this to check label
  it("localized", () => {
    const label = MessageMock.title;
    const props = { ...defaultProps, label, localized: true };
    const { element } = getInputNumberTestable({ ...props }, label);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("disabled", () => {
    const { element, wrapper } = getInputNumberTestable({ disabled: true });
    expect(wrapper.prop("disabled")).toBeTruthy();

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("float", () => {
    const onChange = jest.fn();
    const value = 10.1;
    const { element, wrapper } = getInputNumberTestable({ integer: false, onChange, value });
    expect(wrapper.prop("value")).toEqual(value);
    wrapper.simulate("change", { target: { value: 45.598 } });
    expect(onChange).toHaveBeenCalledWith(45.598);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("min/max", () => {
    const onChange = jest.fn();
    const maxValue = 9;
    const minValue = 5;
    const { element, wrapper } = getInputNumberTestable({ minValue, maxValue, onChange });
    wrapper.simulate("change", { target: { value: maxValue + 1 } });
    expect(onChange).toHaveBeenCalledWith(maxValue);
    wrapper.simulate("change", { target: { value: minValue - 1 } });
    expect(onChange).toHaveBeenCalledWith(minValue);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("numeric values only", () => {
    const onChange = jest.fn();
    const { element, wrapper } = getInputNumberTestable({ onChange });
    wrapper.simulate("change", { target: { value: "text value" } });
    expect(onChange).toHaveBeenCalledWith(null);
    wrapper.simulate("change", { target: { value: null } });
    expect(onChange).toHaveBeenCalledWith(null);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("onChange", () => {
    const onChange = jest.fn();
    const { element, wrapper } = getInputNumberTestable({ onChange });
    wrapper.simulate("change", { target: { value: 1 } });
    expect(onChange).toHaveBeenCalledWith(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("placeholder", () => {
    const placeholder = "Placeholder";
    const { element, wrapper } = getInputNumberTestable({ placeholder });
    expect(wrapper.prop("placeholder")).toEqual(placeholder);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("required", () => {
    const { element, wrapper } = getInputNumberTestable({ required: true });
    expect(wrapper.prop("required")).toBeTruthy();

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("shrink", () => {
    const { element } = getInputNumberTestable({ shrink: false });

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("size", () => {
    const { element, wrapper } = getInputNumberTestable({ size: InputSize.small });
    expect(wrapper.hasClass("MuiInputBase-inputMarginDense"));
    expect(wrapper.hasClass("MuiOutlinedInput-inputMarginDense"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("value", () => {
    const value = 10;
    const { element, wrapper } = getInputNumberTestable({ value });
    expect(wrapper.prop("value")).toEqual(value);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("variant", () => {
    const { element, wrapper } = getInputNumberTestable({ variant: InputVariant.filled });
    expect(wrapper.hasClass("MuiFilledInput-input"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });
});
