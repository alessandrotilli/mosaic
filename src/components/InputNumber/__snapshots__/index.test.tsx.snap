// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber test suite: dataCy 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="custom"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: default 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: disabled 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
      data-cy="input-number"
      disabled={true}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={true}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: float 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value={10.1}
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: localized 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
    data-shrink={true}
  >
    Title
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="intl.message.title"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-42 PrivateNotchedOutline-legendNotched-43"
      >
        <span>
          Title
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: min/max 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: numeric values only 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: onChange 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: placeholder 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Placeholder"
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: required 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={true}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: shrink 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: size 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: value 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value={10}
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-41"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputNumber test suite: variant 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input"
      data-cy="input-number"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="number"
      value=""
    />
  </div>
</div>
`;
